evap/results/exporters.py

```diff
@@ -262,11 +290,17 @@ def write_questionnaire(self, questionnaire, evaluations_with_results, contribut
                 approval_count = 0
 
                 for grade_result in results[questionnaire.id]:
-                    if grade_result.question.id != question.id or not grade_result.has_answers:
+                    if grade_result.question.id != question.id:
+                        continue
+
+                    assert isinstance(grade_result, RatingResult)
+                    if not grade_result.has_answers:
                         continue
+
                     values.append(grade_result.average * grade_result.count_sum)
                     count_sum += grade_result.count_sum
                     if grade_result.question.is_yes_no_question:
+                        assert hasattr(grade_result, "approval_count")
```

Comment:
this is already covered by `assert isinstance(grade_result, RatingResult)` above (same as https://github.com/e-valuation/EvaP/pull/2035#discussion_r1400950031)
```suggestion
```